### PR TITLE
Add support for specifying storage pool when creating a disk.

### PR DESCRIPTION
**Pull request by sizzle0121**
Add field `storage_pool` to `google_compute_disk` resource to support creating disk with the storage pool in which it will reside.
Instead of adding a resource for storage pool and using resourceRef to let disk reference it, we add the field as a string which will contain the URL for a storage pool. The main reason for this decision is that if we add storage pool as a resource, and in the scenario that users want to destroy a storage pool, Terraform will plan to delete all disks first and then delete the storage pool, which we definitely want to prevent from happening.
Therefore, we eventually decided to not support creating storage pool via Terraform, and only support adding the field for disks to specify existing storage pool. The management of storage pool itself will remain through GCP APIs.

```release-note:enhancement
compute: added `storage_pool` field to `google_compute_disk` resource
```

### CODE REVIEW

**Comment by sizzle0121** [comment]
Hi @c2thorn, I think the failed tests are not related to my changes, please review, thanks!

File: mmv1/third_party/terraform/services/compute/resource_compute_disk_test.go.erb [hunk 1]
```diff
@@ -1516,3 +1519,109 @@ resource "google_compute_disk" "foobar" {
...
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+		},
+	})
+
+	cleanupTestingStoragePool(t, storagePoolName)
+}
+
+func setupTestingStoragePool(t *testing.T, storagePoolName string) func() {
```
Comment (c2thorn) [inline mmv1/third_party/terraform/services/compute/resource_compute_disk_test.go.erb:1551]: Should Compute storage pool be a Terraform resource eventually?

**Comment by c2thorn** [comment]
re-running tests to hopefully avoid some of the flakey tests

**Pull request merged**